### ISSUE TITLE
Better documentation, better examples

### ISSUE THREAD

**Issue by Sir4ur0n** [vavr-io/vavr#2234]
Hi,

I use Vavr both at work and on personal projects, and whenever I present Vavr to potential new users (coworkers and friends), the same feedback always comes back within days/weeks: not enough documentation and examples. I heard this sentence tens (if not hundreds) of times. And I have no choice but to agree with them. When I learnt Vavr (Javaslang back in the days) I pretty much dug through all the Javadoc in my IDE to get a grasp at what the various APIs had to offer, because the guide was just not enough.

Figuring out some structures (e.g. `Try`) and how/why to use them looks like it's "left as an exercise for the reader".

The thing is, many (most?) people discovering a new library/paradigm are already going out of their confort zone, so the best way to convince them to start using and getting familiar with Vavr is to take them by the hand. `Try` documentation doesn't even explain `Try` can be a replacement for Java `try/catch` blocks, or provide a side-by-side comparison of `Try` and `try/catch` syntax.

IMHO the easiest way to accompany new comers is to have baby-step documentation and examples. Anyway those who are already familiar with Vavr most likely won't use the guide but directly read the Javadoc, so no loss in expanding.

I also think (but maybe this is another issue?) some Javadocs are too small and could be expanded.

I know you've been working for a few months now on 1.0 and it's a time sink. However I think we could in parallel massively improve the documentation to get newcomers onboard.

The way I see it today, Vavr is an amazing library for Java. An amazing poorly documented library.

If you guys agree, I volunteer to help provide examples/documentation.

**Comment by ruslansennov** [comment]
> If you guys agree, I volunteer to help provide examples/documentation.

Sure, thanks

**Comment by danieldietrich** [comment]
The documentation will get a complete rewrite starting with Vavr 1.
